grpc/src/main/java/io/stargate/grpc/service/Service.java

```diff
@@ -153,12 +152,12 @@ private void handleError(Throwable throwable, StreamObserver<Result> responseObs
   }
 
   private CompletableFuture<Prepared> prepareQuery(
-      Connection connection, String cql, StringValue keyspace, boolean tracing) {
+      Connection connection, String cql, String keyspace, boolean tracing) {
     CompletableFuture<Prepared> future = new CompletableFuture<>();
     final StringBuilder keyBuilder = new StringBuilder();
     connection.loggedUser().ifPresent(user -> keyBuilder.append(user.name()));
-    if (keyspace.isInitialized()) {
-      keyBuilder.append(keyspace.getValue());
+    if (keyspace != null) {
+      keyBuilder.append(keyspace);
```

Comment:
It may be worth using explicit separators between key components to avoid confusing (`key`, `statement`) with (`null`, `keystatement`). It's not likely to manifest, I guess, but might be worth implementing for good measure. WDYT?

Reply:
suggestion: a dedicated `@Value.Immutable` inner class might be as efficient a key as a constructed `String` and will solve the component separation problem.

Reply:
Good suggestion. Let me give that a try.

Reply:
Fixed this.